app/helpers/common/priceFormatter.test.js

```diff
@@ -1,26 +1,59 @@
-import { formatPrice, formatDecimal } from './priceFormatter';
+import { formatPrice, formatNumber } from './priceFormatter';
 
 describe('formatPrice', () => {
   it.each`
     priceValue   | expected
-    ${10}        | ${'10.00'}
-    ${1000}      | ${'1,000.00'}
-    ${999.999}   | ${'999.99'}
+    ${10}       | ${'10.00'}
+    ${1000}     | ${'1,000.00'}
+    ${999.999}   | ${'999.999'}
+    ${999.9999}  | ${'999.9999'}
+    ${999.99999} | ${'1,000.00'}
     ${1234.560}  | ${'1,234.56'}
-    ${12.011}    | ${'12.01'}
-  `('formatPrice $priceValue should be formatted to $expected', ({ priceValue, expected }) => {
-    expect(formatPrice(priceValue)).toEqual(expected);
+    ${12.011}    | ${'12.011'}
+  `('formatPrice({ value: $priceValue }) should be formatted to $expected', ({ priceValue, expected }) => {
+    expect(formatPrice({ value: priceValue })).toEqual(expected);
+  });
+
+  it.each`
+    priceValue   | expected
+    ${99.99}    | ${'99.99'}
+    ${10.001}    | ${'10.00'}
+    ${10.005}    | ${'10.01'}
+    ${10.0005}   | ${'10.00'}
+    ${99.999}    | ${'100.00'}
+  `('formatPrice({ value: $priceValue, maximumFractionDigits: 2 }) should be formatted to $expected', ({ priceValue, expected }) => {
+    expect(formatPrice({ value: priceValue, maximumFractionDigits: 2 })).toEqual(expected);
+  });
+
+  it.each`
+    priceValue   | expected
+    ${99}      | ${'99'}
+    ${10.0}     | ${'10'}
+    ${10.5}     | ${'10.50'}
+    ${10.05}    | ${'10.05'}
+    ${10.0005}  | ${'10.0005'}
+    ${10.00005} | ${'10.0001'}
+  `('formatPrice({ value: $priceValue, maximumFractionDigits: 4, minimumFractionDigits: 0 })'
+  + ' should be formatted to $expected', ({ priceValue, expected }) => {
+    expect(
+      formatPrice({ value: priceValue, maximumFractionDigits: 4, minimumFractionDigits: 0 }),
+    )
+      .toEqual(expected);
```

Comment:
To elaborate on my previous comment about refactoring these tests:

From what I can see, the only difference between each of these three tests are the values of `minimumFractionDigits` and `maximumFractionDigits`. Therefore, we could have a single test with some example cases as follows:

```javascript
describe('formatPrice', () => {
  it.each`
    value        | minimumFractionDigits | maximumFractionDigits | expected
    ${10}        | 0                     | 2                     | ${'10'}
    ${10}        | 2                     | 2                     | ${'10.00'}
    ${999.999}   | 2                     | 4                     | ${'999.99'}
    ${1234.560}  | 2                     | 4                     | ${'1,234.56'}
    ${12.011}    | 2                     | 2                     | ${'12.01'}
  `('formatPrice $value should be formatted to $expected', ({ value, minimumFractionDigits, maximumFractionDigits, expected }) => {
    expect(formatPrice({ value, minimumFractionDigits, maximumFractionDigits })).toEqual(expected);
  });
});
```

(I've not given much thought about the test cases above as they're just there as an example.)

On your issues with alignment, I guess you may not be using a fixed-width font? If not, just make sure that the vertical bars are all in the same column in your editor (usually displayed in the status bar by default in most editors).